config/log-viewer.php

```diff
@@ -73,4 +75,32 @@
         '/vendor/laravel/framework/',
         '/vendor/barryvdh/laravel-debugbar/',
     ],
+
+    /*
+    |--------------------------------------------------------------------------
+    | Log matching patterns
+    |--------------------------------------------------------------------------
+    | Regexes for matching log files
+    |
+    */
+
+    'patterns' => [
+
+        /**
+         * This pattern, used for processing Laravel logs, returns these results:
+         * $matches[0] - the full log line being tested.
+         * $matches[1] - full timestamp between the square brackets (includes microseconds and timezone offset)
+         * $matches[2] - timestamp microseconds, if available
+         * $matches[3] - timestamp timezone offset, if available
+         * $matches[4] - contents between timestamp and the severity level
+         * $matches[5] - environment (local, production, etc)
+         * $matches[6] - log severity (info, debug, error, etc)
+         * $matches[7] - the log text, the rest of the text.
+         */
+        'laravel_regex' => '/^\[(\d{4}-\d{2}-\d{2}[T ]\d{2}:\d{2}:\d{2}\.?(\d{6}([\+-]\d\d:\d\d)?)?)\](.*?(\w+)\.|.*?)('
+            . implode('|', array_filter(Level::caseValues()))
+            . ')?: (.*?)( in [\/].*?:[0-9]+)?$/is',
+
+        'log_match' => '/\[\d{4}-\d{2}-\d{2}[T ]\d{2}:\d{2}:\d{2}(\.\d{6}[\+-]\d\d:\d\d)?\].*/',
```

Comment:
Also consider the fact that the regex patterns now only live in the config file without a fallback.

There's already 100s of installations of this package and some of them have published the current configuration to their app. They won't have these regex patterns in their current configurations, so this would be a breaking change for them.

We should try and not break if possible here. This means reading from the config, but if not found - falling back to a known default.

Reply:
And third, all your commits are coming up as **Unverified** here 🤔 

<img width="886" alt="Screenshot 2022-09-06 at 20 12 58" src="https://user-images.githubusercontent.com/8697942/188698521-d000f5bc-c488-4f8d-bb59-b72e52828da3.png">

Reply:
Hey @arukompas. I've added the naming changes you proposed, makes sense.

With regards to the breaking changes, this is not the one and doesn't need a fallback. Service provider has the config merge section, meaning it will merge the published one with original one to fill out the blanks. Only time it would be a breaking change is when you change something within existing config key as then the published config key would take precedence.

With regards to unverified commits, I must have screwed up GPG somewhere in the process. Nothing concerning.